src/ucp/tag/rndv.c

```diff
@@ -558,15 +689,23 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_matched, (worker, rreq, rndv_rts_hdr),
     rndv_mode = worker->context->config.ext.rndv_mode;
     if (UCP_DT_IS_CONTIG(rreq->recv.datatype)) {
         if (rndv_rts_hdr->address &&
-            (rndv_mode != UCP_RNDV_MODE_PUT_ZCOPY) &&
+            (ucp_rndv_is_get_zcopy(rreq->recv.mem_type, rndv_mode)) &&
             /* is it allowed to use GET Zcopy for the current message? */
             (rndv_rts_hdr->size >= ucp_ep_config(ep)->tag.rndv.min_get_zcopy) &&
             /* is GET Zcopy operation supported? */
             ucp_ep_config(ep)->tag.rndv.max_get_zcopy) {
             /* try to fetch the data with a get_zcopy operation */
             ucp_rndv_req_send_rma_get(rndv_req, rreq, rndv_rts_hdr);
             goto out;
-        } else if (rndv_mode != UCP_RNDV_MODE_GET_ZCOPY) {
+        } else if (rndv_mode == UCP_RNDV_MODE_AUTO) {
```

Comment:
we should handle `if (rndv_mode != UCP_RNDV_MODE_GET_ZCOPY)` separtly for PUT Zcopy.
the current code breaks `UCX_RNDV_SCHEME=put_zcopy` case, because we don't register receiver buffer for it

BTW, why we fallback to RNDV PUT Zcopy scheme in case RNDV pipeline is not needed? I guess we should prefer GET Zcopy. If GET Zcopy RNDV scheme is unsupported (`message size < min_get_zcopy`, requeted RNDV scheme - PUT_ZCOPY, `max_get_zcopy == 0`), we should use PUT Zcopy scheme then

Reply:
@dmitrygx did it address your queries?

Reply:
@bureddy sorry for being late
yes, it is clear fro me now

Reply:
@bureddy but line 710 is not executed for UCX_RNDV_SCHEME=put_zcopy

Reply:
@yosefe thanks. Fixed.

Reply:
so after the fix, UCX_RNDV_SCHEME=put_zcopy would actually **disable** pipelined protocol?

Reply:
yes.